sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -180,6 +180,70 @@ static inline std::string GetHTTPMessagePreBody(Azure::Core::Http::Request const
 
   return httpRequest;
 }
+
+static void CleanupThread()
+{
+  using namespace Azure::Core::Http::_detail;
+  for (;;)
+  {
+    {
+      // Won't continue until the ConnectionPoolMutex is released from MoveConnectionBackToPool
+      std::unique_lock<std::mutex> lockForPoolCleaning(
+          CurlConnectionPool::g_curlConnectionPool.ConnectionPoolMutex);
+      // Wait the defined default time OR to the signal from the conditional variable.
+      // wait_until releases the mutex lock until it wakes up again or it's cancelled.
+      if (CurlConnectionPool::g_curlConnectionPool.ConditionalVariableForCleanThread.wait_for(
+              lockForPoolCleaning,
+              std::chrono::milliseconds(DefaultCleanerIntervalMilliseconds),
+              []() { return CurlConnectionPool::g_curlConnectionPool.ConnectionCounter == 0; }))
```

Comment:
Why are we setting it to 0 before returning? Shouldn't we release the connections from the pool first?

Reply:
not setting to 0, but checking if it is zero.  If it is zero, it means that by the time the thread is waking up, there are no connections in the pool and this lambda return true and the `if` from line 195 is resolved to `true` and goes to line 201 to return and terminate the thread.

Reply:
🤦‍♂️ of course. misread the code.

> If it is zero, it means that by the time the thread is waking up, there are no connections in the pool and this lambda return true and the `if` from line 195 is resolved to `true` and goes to line 201 to return and terminate the thread.

Add this as a comment :)

Something like:
If it is zero, it means that by the time the thread is waking up, there are no connections in the pool and this lambda return true, which means we return and terminate the thread.